### DISCUSSION TITLE
FastDDS 2.9.1 32-bit participant crashes when Shapes Demo is launched in Windows 11

### DISCUSSION THREAD

**Discussion by githubposter12** [eProsima/Fast-DDS#3487] [Unofficial support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

no crash

### Current behavior

I have a trivial fastdds example that creates a domain participant and loops indefinitely.
This is compiled using the Qt IDE with latest MSVC compiler and Ninja build system in 32-bit.
If I run this example, and then launch the FastDDS 2.9.1 Shapes Demo and connect to the same domain,
then my example code will crash, almost every time (see error in log section).

The crash also happens with latest 2.10.1 libraries

### Steps to reproduce

1. run example code
2. launch FastDDS 2.9.1 official shapes demo
3. Options > "Participant Configuration"
4. set domain to same as example code and click "Start"
5. example code crashes

```C++
#include <chrono>
#include <assert.h>
#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
#include <fastdds/dds/domain/DomainParticipantListener.hpp>
#include <fastdds/dds/subscriber/DataReaderListener.hpp>
#include <fastdds/dds/publisher/DataWriterListener.hpp>

using namespace eprosima;
using namespace eprosima::fastdds::dds;
using namespace eprosima::fastrtps::rtps;

class TestCase
{
public:

    DomainParticipant* m_Participant = nullptr;
    DomainParticipantListener* m_ParticipantListener = nullptr;

    TestCase() = default;
    ~TestCase() = default;

    class ParticipantListener : public DomainParticipantListener
    {
    public:
        ParticipantListener()
        {}

        void on_participant_discovery(
            DomainParticipant* participant,
            ParticipantDiscoveryInfo&& info) override {}

        void on_publisher_discovery(
            DomainParticipant* participant,
            WriterDiscoveryInfo&& info) override {}
    };

    void run()
    {
        m_ParticipantListener = new ParticipantListener();

//        StatusMask ParticipantMask = StatusMask::all() >> StatusMask::data_on_readers();
        StatusMask ParticipantMask = StatusMask::all();

        auto factory = DomainParticipantFactory::get_instance();
        m_Participant = factory->create_participant_with_profile("testprofile", m_ParticipantListener, ParticipantMask);
        assert(m_Participant);
        std::cout << "connected." << std::endl;

        while(true)
        {
            std::this_thread::sleep_for(std::chrono::milliseconds(1000));
        }
    }

};

int main()
{
    TestCase test;
    test.run();

    return 0;
}
```

```cmake
cmake_minimum_required(VERSION 3.14)

project(TestCase LANGUAGES CXX)

set(CMAKE_CXX_STANDARD 20)
set(CMAKE_CXX_STANDARD_REQUIRED ON)

# Find FastDDS requirements
if(NOT DEFINED ENV{FASTRTPSHOME})
    message(FATAL_ERROR "You must set FASTRTPSHOME environment variable!")
endif()
if(NOT fastcdr_FOUND)
    find_package(fastcdr REQUIRED)
endif()
if(NOT fastrtps_FOUND)
    find_package(fastrtps REQUIRED)
endif()

add_executable (bug1
    bug1.cpp
    DEFAULT_FASTRTPS_PROFILES.xml
)
target_include_directories(bug1
    PUBLIC
        include
        $ENV{FASTRTPSHOME}/include
)
target_link_libraries(bug1
    PUBLIC
        fastrtps fastcdr)
```

### Fast DDS version/commit

2.9.1 official pre-compiled library
2.9.1 official Shaped Demo

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

UDPv4

### Additional context

Windows 11 64-bit Version 21H2 OS Build 22000.1696
Visual Studio 2022 17.5.4
Qt Creator 10.0.0
Ninja 1.11.1

### XML configuration file

```xml
<?xml version="1.0" encoding="UTF-8" ?>
<profiles xmlns="http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles">

    <participant profile_name="testprofile">
            <domainId>2</domainId>
        <rtps>
            <name>testprofile</name>
        </rtps>
    </participant>

</profiles>
```

### Relevant log output

```shell
connected.
Assertion failed: (ctrl_data->m_value_bytes % table.size) == 0, file C:\FIW\src\fastrtps\thirdparty\boost\include\boost/interprocess/segment_manager.hpp, line 861
Press <RETURN> to close this window...
```

**Comment by Mario-DL** [comment]
Hi @githubposter12,

Thanks for the detailed description. According to our current [Platform Support](https://github.com/eProsima/Fast-DDS/blob/master/PLATFORM_SUPPORT.md), neither Windows 11 nor MSVC 2022 compiler are an officially supported platforms, so I am moving this issue to the corresponding discussion forum.

Please, re-check it with an official platform to see if the issue persists. In addition, would the fact that one application is being compiled with a 32-bit compiler while the other one is 64-bit and both are launched within a 64 bit environment be related to the issue ?

**Reply by songyuc** [reply]
Hi, is there any plan for support on Win11?
